### DISCUSSION TITLE
Create shipment operation from MerchantFulfillment

### DISCUSSION THREAD

**Discussion by Lemodo20** [abuzuhri/Amazon-SP-API-CSharp#489] [General]
Hi,

i try to generate DHL Shipping Label with your function "Create shipment operation from MerchantFulfillment".
When i run the code, it gives me the error "FikaAmazonAPI.AmazonSpApiSDK.Models.Exceptions.AmazonInvalidInputException: "1 validation error detected: Value 'DHL Paket National' at 'shippingServiceId' failed to satisfy constraint: Member must satisfy regular expression pattern: [\S]+"

I replace the "UPS_PTP_2ND_DAY_AIR" with "DHL Paket Nation". Where can i find the correct values for this field? Also for "shippingServiceOfferID"?

**Comment by RenzoF** [comment]
Hi,
I take it you are trying to buy shipping using amazon shipping? if so try getting the available rates you have you should the see the shipment Id in the results there

**Comment by Lemodo20** [comment]
As we use our own DHL contract, i don't have to request for available rates!? But i will try to find out, how to get the available rates. Thank you for the hint

**Reply by RenzoF** [reply]
You do if you are trying to get your labels from amazon, if you are getting your labels from DHL and you are trying to mark the order as shipped then you need to use the POST_ORDER_FULFILLMENT_DATA feed

**Reply by RenzoF** [reply]
by the way, the request available rates is just to see what shippingId they have for DHL, once you know that you don't need to request rates each time

**Reply by Lemodo20** [reply]
Have to create the labels over Amazon, but Amazon use our DHL account and we them charged from DHL, not from Amazon. 

For "normal" shippings i use POST_FLAT_FILE_FULFILLMENT_DATA. This should be similar like POST_ORDER_FULFILLMENT_DATA, but as flatfile (can use same files as i used before with MWS)

**Reply by Lemodo20** [reply]
I looked now arround... Where can i find the function for the available rates?
